### PR TITLE
Fix the GetTrimmedField and GetTrimmedLine functions

### PR DESCRIPTION
**Pull request by elviscapiaq**
- The function GetTrimmedLine is  not considering new lines characters `/n`.
- Also, the GetTrimmedField does not check delimitiers characters inside quotes.

### CODE REVIEW

**Comment by angela28chen** [comment]
Can you add unit tests for this PR? I believe dive core unit tests are under `dive_core/tests`. Specifically the files for `string_utils_test.cpp` and `available_metrics_test.cpp` exist already but the testcases should be updated to cover the edge case you mentioned. Optional but it's probably also a good idea to add other unit tests for the utils function if you can think of any.

**Comment by angela28chen** [comment]
Just an idea but what about moving the stringutils library out from `dive_core/common` and into `utils/`? It seems like a more logical place to put it now that we have a separate utilities dir. The other stuff under dive_core/common seems to be more specialized but "StringUtils" seems widely applicable

File: utils/string_utils.cpp [hunk 1]
```diff
@@ -40,18 +40,117 @@ void RemoveQuotes(std::string& s)
...
+            {
+                break;
+            }
+            // Add the character (it's part of an unquoted field)
+            field += c;
+        }
+    }
+
     Trim(field);
     RemoveQuotes(field);
```
[the remark this answers is not in the document]
Comment (elviscapiaq) [reply utils/string_utils.cpp]: You are right, but in the new version I am using it.

**Comment by elviscapiaq** [comment]
> Just an idea but what about moving the stringutils library out from `dive_core/common` and into `utils/`? It seems like a more logical place to put it now that we have a separate utilities dir. The other stuff under dive_core/common seems to be more specialized but "StringUtils" seems widely applicable

Thanks for pointing this out. I added unit tests and also move out the `StringUtils` to `utils/`.

File: utils/string_utils_test.cpp [hunk 2]
```diff
@@ -360,7 +360,7 @@ TEST(StringUtils, GetTrimmedField_QuotedSimple)
 TEST(StringUtils, GetTrimmedField_QuotedWithDelimiters)
 {
     // Delimiters inside quotes should be ignored
-    std::stringstream ss("\"Doe, John\",30,\"New York, NY\"");
+    std::stringstream ss("  \"Doe, John\"  ,30,\"New York, NY\"");
```
[the remark this answers is not in the document]
Comment (angela28chen) [reply utils/string_utils_test.cpp]: I understand that the comma between Doe and John is expected to be ignored, but why is this PR adding additional spacing to this test case? I don't quite follow
